File: leetcode/solutions/1008.ConstructBinarySearchTreeFromPreorderTraversal/bstFromPreorder_test.go
package solution

import (
	"testing"

	"github.com/google/go-cmp/cmp"
	ds "github.com/yukisato/leetcode-util/datastructure"
)

func Test_1008_ConstructBinarySearchTreeFromPreorderTraversal(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  []int
		out *TreeNode
	}{
		{
			[]int{8, 5, 1, 10, 12},
			ds.CreateTreeNodeFromString([]string{"8", "5", "10", "1", "null", "null", "12"}),
		},
		{
			[]int{8, 5, 1, 7, 10, 12},
			ds.CreateTreeNodeFromString([]string{"8", "5", "10", "1", "7", "null", "12"}),
		},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := bstFromPreorder(tt.in)

			if !cmp.Equal(got, tt.out) {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/1008.ConstructBinarySearchTreeFromPreorderTraversal/bstFromPreorder_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import bstFromPreorder from "./bstFromPreorder.ts";
import {
  BinaryTreeNode,
  createBinaryTreeNode,
} from "../../../datastructure/BinaryTreeNode.ts";

Deno.test("1008. Construct Binary Search Tree from Preorder Traversal", () => {
  type test = [number[], BinaryTreeNode<number> | null];
  const tests: test[] = [
    // [
    //   [8,5,1],
    //   createBinaryTreeNode([8,5,null,1]),
    // ],
    [[8, 5, 1, 7, 10, 12], createBinaryTreeNode([8, 5, 10, 1, 7, null, 12])],
  ];

  for (const [arg, want] of tests) {
    assertEquals(bstFromPreorder(arg), want);
  }
});
